### PR TITLE
refactor(redux-async-flows): adding typescript support - WIP

### PR DESCRIPTION
**Pull request by alancampora**
# Typescript support
This is still a work in progress PR. 

## Purpose 
Using types will allow to have a better understanding of the API. 
Before investing time on it, do you think it worth it @ssaucedo 

## Details
* Adding typescript package
* Adding types to index.ts, removing index.js
* Adding tsconfig file
* It would be possible to remove babel (?)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+interface Register {
+  [index:string]: Array<Function>
+}
+
+interface Action {
+  type: string
+  payload: object
+}
+
+interface Output{
```
[the remark this answers is not in the document]
Comment (ssaucedo) [reply src/index.ts]: works for me!

**Comment by ssaucedo** [comment]
@alancampora sure it makes sense. Thanks!

**Comment by ssaucedo** [comment]
let's submit pull requests against development branch

**Comment by alancampora** [comment]
re-doing pr from this repo

**Pull request closed without merging**